src/game/game.go

```diff
@@ -1,38 +1,179 @@
+	var coordsMovableTiles []Coords
+
+	coordsEmptyTile, err := findEmptyTile(grid)
+	if err != nil {
+		return coordsMovableTiles
+	}
+
+	size := byte(len(grid))
+	if coordsEmptyTile.y-1 != 255 {
+		coordsMovableTiles = append(coordsMovableTiles, Coords{coordsEmptyTile.y - 1, coordsEmptyTile.x})
+	}
+	if coordsEmptyTile.x+1 < size {
+		coordsMovableTiles = append(coordsMovableTiles, Coords{coordsEmptyTile.y, coordsEmptyTile.x + 1})
+	}
+	if coordsEmptyTile.y+1 < size {
+		coordsMovableTiles = append(coordsMovableTiles, Coords{coordsEmptyTile.y + 1, coordsEmptyTile.x})
+	}
+	if coordsEmptyTile.x-1 != 255 {
+		coordsMovableTiles = append(coordsMovableTiles, Coords{coordsEmptyTile.y, coordsEmptyTile.x - 1})
+	}
+	return coordsMovableTiles
+}
+
+func CoordsFromDirection(grid Grid, dir rune) (Coords, error) {
+	var coordsMovableTiles Coords
+
+	coordsEmptyTile, err := findEmptyTile(grid)
+	if err != nil {
+		return coordsEmptyTile, err
+	}
+
+	size := byte(len(grid))
+	if dir == 's' || dir == 'S' {
+		if coordsEmptyTile.y-1 != 255 {
+			coordsMovableTiles.y = coordsEmptyTile.y - 1
+			coordsMovableTiles.x = coordsEmptyTile.x
+		} else {
+			err = errors.New("It's not possible to move 'bottom'")
+		}
+	} else if dir == 'q' || dir == 'Q' {
+		if coordsEmptyTile.x+1 < size {
+			coordsMovableTiles.y = coordsEmptyTile.y
+			coordsMovableTiles.x = coordsEmptyTile.x + 1
+		} else {
+			err = errors.New("It's not possible to move 'left'")		
+		}
+	} else if dir == 'z' || dir == 'Z' {
+		if coordsEmptyTile.y+1 < size {
+			coordsMovableTiles.y = coordsEmptyTile.y + 1
+			coordsMovableTiles.x = coordsEmptyTile.x
+		} else {
+			err = errors.New("It's not possible to move 'top'")
+		}
+	} else if dir == 'd' || dir == 'D' {
+		if coordsEmptyTile.x-1 != 255 {
+			coordsMovableTiles.y = coordsEmptyTile.y
+			coordsMovableTiles.x = coordsEmptyTile.x - 1
+		} else {
+			err = errors.New("It's not possible to move 'right'")
+		}
+	}
+	if err != nil {
+		return coordsMovableTiles, err	
+	}
+	return coordsMovableTiles, nil
+}
+
+func isTileInMovableTiles(grid Grid, coordsTileToMove Coords) bool {
+	for _, value := range ListMovableTiles(grid) {
+		if AreCoordsEquals(value, coordsTileToMove) {
+			return true
+		}
+	}
+	return false
+}
+
+func Move(grid Grid, coordsTileToMove Coords) (Grid, error) {
+	if !isTileInMovableTiles(grid, coordsTileToMove) {
+		return grid, errors.New(fmt.Sprintf("The tile ate coords (%d, %d) is not movable", coordsTileToMove.y, coordsTileToMove.y))
+	}
+
+	emptyCoords, err := findEmptyTile(grid)
+	if err != nil {
+		return grid, err
+	}
+
+	newCoords, err := findTileByValue(grid, grid[coordsTileToMove.y][coordsTileToMove.x])
+	if err != nil {
+		return grid, err
+	}
+
+	newGrid := DeepCopyGrid(grid)
+	newGrid[emptyCoords.y][emptyCoords.x], newGrid[newCoords.y][newCoords.x] = grid[newCoords.y][newCoords.x], grid[emptyCoords.y][emptyCoords.x]
+	return newGrid, nil
+}
+
+func MoveByValue(grid Grid, value byte) (Grid, error) {
```

Comment:
You just use it in tests. Remove all dead code. We'll add it only when required.